common/flatpak-installation.c

```diff
@@ -1221,11 +1222,41 @@ flatpak_installation_list_remotes_by_type (FlatpakInstallation     *self,
   if (!flatpak_dir_maybe_ensure_repo (dir_clone, cancellable, error))
     return NULL;
 
+#if OSTREE_CHECK_VERSION (2018, 9)
+  OstreeRepo *repo = flatpak_dir_get_repo (dir_clone);
+  if (repo != NULL)
+    default_repo_finders = ostree_repo_get_default_repo_finders (repo);
+#endif
+
+  /* If NULL or an empty array of types is passed then we use the default set
+   * provided by ostree, or fall back to using all */
   for (i = 0; i < NUM_FLATPAK_REMOTE_TYPES; ++i)
     {
-      /* If NULL or an empty array of types is passed then we include all types */
-      types_filter[i] = (num_types == 0) ? TRUE : FALSE;
+      if (num_types != 0)
+        types_filter[i] = FALSE;
+#if OSTREE_CHECK_VERSION (2018, 9)
+      else if (default_repo_finders == NULL)
+        types_filter[i] = TRUE;
+#endif
+    }
+
+#if OSTREE_CHECK_VERSION (2018, 9)
+  if (default_repo_finders != NULL && num_types == 0)
+    {
+      g_autofree char *default_repo_finders_str = g_strjoinv (" ", default_repo_finders);
+      g_debug ("Using default repo finder list: %s", default_repo_finders_str);
+
+      if (g_strv_contains (default_repo_finders, "config"))
+        types_filter[FLATPAK_REMOTE_TYPE_STATIC] = TRUE;
+      else if (g_strv_contains (default_repo_finders, "lan"))
+        types_filter[FLATPAK_REMOTE_TYPE_LAN] = TRUE;
+      else if (g_strv_contains (default_repo_finders, "mount"))
+        types_filter[FLATPAK_REMOTE_TYPE_USB] = TRUE;
+      else
+        g_warning ("Unknown value in list returned by ostree_repo_get_default_repo_finders(): %s",
```

Comment:
Not sure this should be a `g_warning()`. What’s the harm in ignoring a default repo finder which flatpak doesn’t support?

Reply:
Also, this check for an unknown value won’t work properly if the `default_repo_finders` list contains at least one known value. It needs to be changed to an iteration over the list.

Reply:
Sure, we can change it back to a `g_debug`. I didn't think about the possibility of libostree gaining a finder that flatpak wouldn't know about.